src/prefect/client/base.py

```diff
@@ -246,11 +243,21 @@ async def _send_with_retry(
         return response
 
     async def send(self, *args, **kwargs) -> Response:
+        """
+        Send a request with automatic retry behavior for the following status codes:
+
+        - 423 Resource locked
+        - 429 CloudFlare-style rate limiting
+        - 502 Bad Gateway
+        - 503 Service unavailable
+        """
+
         api_request = partial(super().send, *args, **kwargs)
 
         response = await self._send_with_retry(
             request=api_request,
             retry_codes={
+                status.HTTP_423_LOCKED,
```

Comment:
@bunchesofdonald, will this play well with concurrency limits?  I know we're using 423 when there are no more slots left.  Since concurrency limits have their own wait/retry logic, we wouldn't want to have the underlying client retry too, right?

Reply:
Ah yea, let me know if that's a problem - we don't strictly need this new 423 behavior in our base client, I just implemented it since I was looking at our status code usage in the backend and this was more appropriate for one of our use cases

Reply:
I like the usage!  We just might need to adjust the concurrency limits implementation

Reply:
Yeah, it'll cause a problem. The issue I see is that the client retry logic only has a set number of retries, whereas the concurrency limits should retry indefinitely.

Reply:
I could change the concurrency implementation to use a `200` with status information in the body, that someone could add 423 to the retries and break concurrency seems like a 'bug'.

Reply:
I like 423 a lot for the concurrency case, though, it's very close to the heart of what it's about.  I'd rather see if we can have the client-side concurrency implementation just tell the HTTP client that it is handling 423s somehow.  It seems generally useful to allow for adjusting the retry statuses on a call-by-call basis if we need to.

Looking at the client, we [currently have this implementation](https://github.com/PrefectHQ/prefect/blob/main/src/prefect/client/orchestration.py#L2540-L2546), which will use the default retry logic, trying several times before raising, which will _then_ get caught by the outer concurrency implementation for _its_ retry logic:

```python
    async def increment_concurrency_slots(
        self, names: List[str], slots: int, mode: str
    ) -> httpx.Response:
        return await self._client.post(
            "/v2/concurrency_limits/increment",
            json={"names": names, "slots": slots, "mode": mode},
        )
```

What if we had a way to adjust that on a call-by-call basis, like:

```python
    async def increment_concurrency_slots(
        self, names: List[str], slots: int, mode: str
    ) -> httpx.Response:
        async with self._suppress_retries_for(423):
            return await self._client.post(
                "/v2/concurrency_limits/increment",
                json={"names": names, "slots": slots, "mode": mode},
            )
```

It's a little messy but it would give us some super targeted behavior tweaks when we need it

Reply:
I could get behind something like that.

Reply:
Alternatively I could expose the retry codes interface to the `post` method of the client, and implement it such that 502 and 503 are always retried, and additional codes can be provided.  I can default those additional codes to 423 and 429 but if others are provided we respect that.

Also happy to simplify the PR and defer this decision if it's a distraction

Reply:
I was thinking about `kwargs` too, but I wasn't sure the best way to override the HTTP methods on the underlying client.  Honestly, we should probably migrate our retry implementation to a _transport_ class instead of subclassing the HTTP client itself, then I believe it would be more straightforward